File: backend/Configure.cs


using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using System.Threading.Tasks;

namespace backend
{
    public class Startup
    {
        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
        {
            try
            {
                if (env.IsDevelopment())
                {
                    app.UseDeveloperExceptionPage();
                }

                app.UseRouting();

                app.UseEndpoints(endpoints =>
                {
                    endpoints.MapGet("/", async context =>
                    {
                        await context.Response.WriteAsync("Hello World!");
                    });
                });
            }
            catch (Exception ex)
            {
                app.Use(async (context, next) =>
                {
                    await context.Response.WriteAsync($"An error occurred: {ex.Message}");
                });
            }
        }
    }
}

File: frontend/components/Configure.js


import React, { useEffect } from 'react';

const Configure = ({ app, env }) => {
  useEffect(() => {
    const configureRequestPipeline = async () => {
      if (env === 'Development') {
        app.use(async (context, next) => {
          await context.response.writeAsync('Hello World!');
          await next();
        });
      } else {
        app.use(async (context, next) => {
          await context.response.writeAsync('Hello World!');
          await next();
        });
      }
    };

    configureRequestPipeline();
  }, [app, env]);

  return <div />;
};

export default Configure;